service/storage_proxy.cc

```diff
@@ -210,6 +210,36 @@ class storage_proxy::remote {
 
     bool _stopped{false};
 
+    // Transform a read command by reversing the schema version and the bounds of the clustering key ranges. The transformation
+    // is done only for reverse read queries in a mixed node cluster when the native_reverse_queries feature flag is turned off.
+    query::read_command transform_to_from_legacy_format_for_native_reversed_queries(query::read_command cmd, bool reversed)
+    {
+        if (!reversed || _sp.features().native_reverse_queries)
+            return cmd;
+
+        auto schema = local_schema_registry().get(cmd.schema_version)->get_reversed();
+        cmd.schema_version = schema->version();
+        cmd.slice = query::legacy_reverse_slice_to_native_reverse_slice(*schema, cmd.slice);
+
+        return cmd;
+    }
+
+    // Transform a reconcilable_result by reversing stored mutations. The transformation is done only for reverse read queries
+    // in a mixed node cluster when the native_reverse_queries feature flag is turned off.
+    reconcilable_result transform_to_from_legacy_format_for_native_reversed_queries(reconcilable_result result, bool reversed)
+    {
+        if (!reversed || _sp.features().native_reverse_queries)
+            return result;
+
+        for (auto& partition : result.partitions()) {
+            auto& m = partition.mut();
+            auto schema = local_schema_registry().get(m.schema_version());
+            m = frozen_mutation(reverse(m.unfreeze(schema)));
+        }
+
+        return result;
+    }
+
```

Comment:
This patch should come earlier in the series. When you remove support for half-reversed mutations in one place, insert a call to one of these at its call site, thus showing how you maintain compatibility, rather than removing compatibility and re-adding it later in separate places.